Documentation/gettingstarted/tls-visibility.rst

```diff
@@ -0,0 +1,308 @@
+the connection as invalid.  In this model, the network firewall uses the certificate signed by the internal
+CA to act like the destination service and terminate the TLS connection.  This allows the firewall to
+inspect and even modify the application layer data, and then initiate another TLS connect to the actual
+destination service.
+
+The CA model within TLS is based on cryptographic keys and certificates.  Realizing the above model
+requires four primary steps:
+
+1) Create an internal certificate authority by generating a CA private key and CA certificate.
+
+2) For any destination where TLS inspection is desired (e.g., swapi.co in the example below), generate a
+private key and certificate signing request with a common name that matches the destination DNS name.
+
+3) Use the CA private key to create a signed certificate.
+
+4) Ensure that all clients where TLS inspection is have the CA certificate installed so that they will
+trust all certificates signed by that CA.
+
+5) Given that Cilium will be terminating the initial TLS connection from the
+client and creating a new TLS connection to the destination, Cilium must be told the set of CAs that it
+should trust when validating the new TLS connection to the destination service.
+
+.. note::
+
+    In a non-demo environment it is EXTREMELY important that you keep the above private keys safe, as anyone
+    with access to this private key will be able to inspect TLS-encrypted traffic (certificates on the other
+    hand are public information, and are not at all sensitive).  In the guide below, the
+    CA private key does not need to be provided to Cilium at all (it is used only to create certificates, which
+    can be done offline) and private keys for individual destination services are stored as Kubernetes secrets.
+    These secrets should be stored in a namespace where they can be accessed by Cilium, but not general purpose
+    workloads.
+
+Generating and Installing TLS Keys and Certificates
+===================================================
+
+Now that we have explained the high-level certificate model used by TLS, we will walk through the
+concrete steps to generate the appropriate keys and certificates using the ``openssl`` utility.
+
+TODO:  install instructions for openssl cmd-line utility?
+
+1) Create internal Certificate Authority.
+
+Generate CA private key named 'myCA.key':
+
+.. parsed-literal::
+
+    $ openssl genrsa -des3 -out myCA.key 2048
+
+Enter any password, just remember it for some of the later steps.
+
+Generate CA certificate from the private key:
+
+.. parsed-literal::
+
+    $ openssl req -x509 -new -nodes -key myCA.key -sha256 -days 1825 -out myCA.crt
+
+The values you enter for each prompt do not need to be any specific value, and do not need to be
+accurate.
+
+2) Generate an internal private key and certificate signing with a common name that matches the DNS name
+of the destination service to be intercepted for inspection.
```

Comment:
`of the destination service to be intercepted for inspection.` it would be worth to have an example of such DNS name

Reply:
agreed.